improver/calibration/reliability_calibration.py

```diff
@@ -712,23 +996,22 @@ def process(self, forecast, reliability_table):
         Args:
             forecast (iris.cube.Cube):
                 The forecast to be calibrated.
-            reliability_table (iris.cube.Cube):
+            reliability_table (iris.cube.Cube or iris.cube.CubeList):
                 The reliability table to use for applying calibration.
         Returns:
             calibrated_forecast (iris.cube.Cube):
                 The forecast cube following calibration.
         """
-        self._threshold_coords_equivalent(forecast, reliability_table)
         self.threshold_coord = find_threshold_coordinate(forecast)
 
         forecast_thresholds = forecast.slices_over(self.threshold_coord)
-        reliability_thresholds = reliability_table.slices_over(self.threshold_coord)
-        slices = zip(forecast_thresholds, reliability_thresholds)
 
         uncalibrated_thresholds = []
         calibrated_cubes = iris.cube.CubeList()
-        for forecast_threshold, reliability_threshold in slices:
-
+        for forecast_threshold in forecast_thresholds:
+            reliability_threshold = self._extract_matching_reliability_table(
```

Comment:
Codacy found an issue: [No value for argument 'reliability_table' in staticmethod call](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6001543)